02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,451 @@
+Therefore, where we have multi-mapped zebrafish Ensembl gene IDs, we will take the mappings with more than one database to support the assertion.
+
+Before we do, let's take a look how many multi-mapped genes there are in the data frame.
+
+```{r}
+human_zebrafish_mapped_df %>%
+  # Group by human gene symbols
+  dplyr::group_by(human_symbol) %>%
+  # Count the number of rows in the dataframe for each symbol
+  dplyr::count() %>%
+  # Filter out the symbols without multimapped genes
+  dplyr::filter(n > 1)
+```
+Looks like we have 4,192 human gene symbols with multiple mappings.
+
+Now let's filter out the less reliable mappings.
+
+```{r}
+filtered_zebrafish_ensembl_df <- human_zebrafish_mapped_df %>%
+  # Count the number of databases in the support column for each prediction
+  dplyr::mutate(n_databases = stringr::str_count(support, ",") + 1) %>%
+  # Group by human gene symbols
+  dplyr::group_by(human_symbol) %>%
+  # Now filter for the rows with more than one database in support for each human gene symbol
+  dplyr::filter(n_databases > 1)
+
+head(filtered_zebrafish_ensembl_df)
+```
+
+Let's count how many multi-mapped genes we have now.
+
+```{r}
+filtered_zebrafish_ensembl_df %>%
+  # Group by human gene symbols
+  dplyr::group_by(human_symbol) %>%
+  # Count the number of rows in the dataframe for each symbol
+  dplyr::count() %>%
+  # Filter out the symbols without multimapped genes
+  dplyr::filter(n > 1)
+```
+Now we only have 1,803 multi-mapped genes, compared to the 4,192 that we began with.
+Although we haven't filtered down to zero multi-mapped genes, we have hopefully removed some of the less _reliable_ mappings.
+
+### Write results to file
+
+Now let's write our `filtered_zebrafish_ensembl_df` object, with the _reliable_ zebrafish Ensembl IDs for each unique human gene symbol, to file.
+
+```{r}
+readr::write_tsv(
+  filtered_zebrafish_ensembl_df,
+  file.path(
+    results_dir,
+    # Replace with a relevant output file name
+    "GSE71270_filtered_zebrafish_ensembl_to_human_gene_symbol.tsv"
+  )
+)
+```
+
+# Resources for further learning
+
+- [Genenames.org: the HGNC resources in 2015](https://pubmed.ncbi.nlm.nih.gov/25361968/) [@Gray2015]
+- [HCOP: the HGNC comparison of orthology predictions search tool](https://pubmed.ncbi.nlm.nih.gov/16284797/) [@Wright2005]
+- See more on [HCOP and interpreting the data in HGNC](https://www.genenames.org/help/hcop/)[@hcop-help]
+- See our [RNA-seq ortholog mapping notebook](https://alexslemonade.github.io/refinebio-examples/03-rnaseq/ortholog-mapping_rnaseq_01_ensembl.Rmd) as another applicable example, since the steps for this workflow do not change with technology. 
+
+# Session info
+
+At the end of every analysis, before saving your notebook, we recommend printing out your session info. 
+This helps make your code more reproducible by recording what versions of software and packages you used to run this. 
+
+```{r}
+# Print session info
+sessioninfo::session_info()
+```
```

Comment:
It won't let me add a line because of the backticks here, but here we will need to add a new line and then `# References` and a new line. In accordance with #301